ServerRequestContext can not be used as parameter of ServerExceptionMapper

### Describe the bug

The documentation for `ServerExceptionMapper` https://quarkus.io/guides/rest#exception-mapping states:
> Your exception mapper may declare any of the following parameter types:
> [...}
> Any of the [Context objects](https://quarkus.io/guides/rest#context-objects)

The list of context objects also includes the [ServerRequestContext](https://javadoc.io/doc/io.quarkus.resteasy.reactive/resteasy-reactive/3.21.2/org/jboss/resteasy/reactive/server/spi/ServerRequestContext.html).

When I want to use the ServerRequestContext as method parameter, I get an exception on startup, see below.

fails:
```
    @ServerExceptionMapper
    RestResponse<String> handle(RuntimeException e, ServerRequestContext serverRequestContext) {
        return RestResponse.ok(serverRequestContext.getRequestHeaders().getRequestHeaders().toString());
    }
```

It works perfectly if I instead use an `@Inject`:
 ```
    @Inject
    ServerRequestContext serverRequestContext;

    @ServerExceptionMapper
    RestResponse<String> handle(RuntimeException e) {
        return RestResponse.ok(serverRequestContext.getRequestHeaders().getRequestHeaders().toString());
    }
```

### Expected behavior

I can use any of the context objects https://quarkus.io/guides/rest#context-objects as method parameter; Since that is what the doc says.

### Actual behavior

```
2025-04-11 20:57:12,481 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (vert.x-worker-thread-1) Failed to start quarkus [Error Occurred After Shutdown]: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#handleClassLevelExceptionMappers threw an exception: java.lang.RuntimeException: Parameter 'serverRequestContext' of method 'handle of class 'org.acme.GreetingResource' is not allowed
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.getTargetMethodParamsInfo(ServerExceptionMapperGenerator.java:608)
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.generateRRResponse(ServerExceptionMapperGenerator.java:479)
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.generatePerClassMapper(ServerExceptionMapperGenerator.java:158)
        at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.handleClassLevelExceptionMappers(ResteasyReactiveProcessor.java:396)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:372)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.reloadExistingApplication(AugmentActionImpl.java:303)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.reloadExistingApplication(AugmentActionImpl.java:61)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:200)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.restartCallback(IsolatedDevModeMain.java:183)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:555)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:455)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$6.call(VertxHttpHotReplacementSetup.java:163)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$6.call(VertxHttpHotReplacementSetup.java:150)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$4(ContextImpl.java:192)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)
        at io.vertx.core.impl.ContextImpl$1.execute(ContextImpl.java:221)
        at io.vertx.core.impl.WorkerTask.run(WorkerTask.java:56)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#handleClassLevelExceptionMappers threw an exception: java.lang.RuntimeException: Parameter 'serverRequestContext' of method 'handle of class 'org.acme.GreetingResource' is not allowed
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.getTargetMethodParamsInfo(ServerExceptionMapperGenerator.java:608)
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.generateRRResponse(ServerExceptionMapperGenerator.java:479)
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.generatePerClassMapper(ServerExceptionMapperGenerator.java:158)
        at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.handleClassLevelExceptionMappers(ResteasyReactiveProcessor.java:396)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)

        at io.quarkus.builder.Execution.run(Execution.java:122)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:78)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:161)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:368)
        ... 21 more
Caused by: java.lang.RuntimeException: Parameter 'serverRequestContext' of method 'handle of class 'org.acme.GreetingResource' is not allowed      
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.getTargetMethodParamsInfo(ServerExceptionMapperGenerator.java:608)
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.generateRRResponse(ServerExceptionMapperGenerator.java:479)
        at org.jboss.resteasy.reactive.server.processor.generation.exceptionmappers.ServerExceptionMapperGenerator.generatePerClassMapper(ServerExceptionMapperGenerator.java:158)
        at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.handleClassLevelExceptionMappers(ResteasyReactiveProcessor.java:396)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)
```

### How to Reproduce?

Download Reproducer: 
[rest-inject-server-req-context.zip](https://github.com/user-attachments/files/19713010/rest-inject-server-req-context.zip)
mvn quarkus:dev
Error from above happens directly

### Output of `uname -a` or `ver`

MINGW64_NT-10.0-22621 THEROCK 3.1.7-340.x86_64 2021-03-26 22:17 UTC x86_64 Msys

### Output of `java -version`

openjdk 17.0.1 2021-10-19 OpenJDK Runtime Environment Temurin-17.0.1+12 (build 17.0.1+12) OpenJDK 64-Bit Server VM Temurin-17.0.1+12 (build 17.0.1+12, mixed mode, sharing)

### Quarkus version or git rev

3.21.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven home: C:\tools\java\maven Java version: 17.0.1, vendor: Eclipse Adoptium, runtime: C:\tools\java\17-temurin Default locale: de_DE, platform encoding: Cp1252 OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customexceptions/MyOtherExceptionMapper.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customexceptions/MyOtherExceptionMapper.java
@@ -4,6 +4,7 @@ import jakarta.ws.rs.core.Response;
 
 import org.jboss.resteasy.reactive.server.ServerExceptionMapper;
 import org.jboss.resteasy.reactive.server.SimpleResourceInfo;
+import org.jboss.resteasy.reactive.server.spi.ServerRequestContext;
 
 import io.vertx.ext.web.RoutingContext;
 
@@ -17,7 +18,9 @@ public class MyOtherExceptionMapper {
 
     @ServerExceptionMapper
     public Response handleMyOtherException(RoutingContext routingContext, MyOtherException myOtherException,
-            SimpleResourceInfo simplifiedResourceInfo) {
+            SimpleResourceInfo simplifiedResourceInfo, ServerRequestContext serverRequestContext) {
+        serverRequestContext.getResteasyReactiveResourceInfo();
+        simplifiedResourceInfo.getMethodName();
         return Response.status(411).build();
     }
 }

--- a/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/CustomContainerRequestFilter.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/CustomContainerRequestFilter.java
@@ -5,12 +5,15 @@ import jakarta.ws.rs.core.HttpHeaders;
 import jakarta.ws.rs.core.UriInfo;
 
 import org.jboss.resteasy.reactive.server.ServerRequestFilter;
+import org.jboss.resteasy.reactive.server.spi.ServerRequestContext;
 
 public class CustomContainerRequestFilter {
 
     @ServerRequestFilter
-    public void whatever(UriInfo uriInfo, HttpHeaders httpHeaders, ContainerRequestContext requestContext) {
-        String customHeaderValue = uriInfo.getPath() + "-" + httpHeaders.getHeaderString("some-input");
+    public void whatever(UriInfo uriInfo, HttpHeaders httpHeaders, ContainerRequestContext requestContext,
+            ServerRequestContext serverRequestContext) {
+        String customHeaderValue = uriInfo.getPath() + "-" + httpHeaders.getHeaderString("some-input") + "-"
+                + serverRequestContext.getRequestHeaders().getHeaderString("some-other-input");
         requestContext.getHeaders().putSingle("custom-header", customHeaderValue);
     }
 }

--- a/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/CustomFiltersTest.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/CustomFiltersTest.java
@@ -31,14 +31,16 @@ public class CustomFiltersTest {
 
     @Test
     public void testFilters() {
-        Headers responseHeaders = RestAssured.given().header("some-input", "bar").get("/custom/req")
-                .then().statusCode(200).body(Matchers.containsString("/custom/req-bar-null")).extract().headers();
+        Headers responseHeaders = RestAssured.given().header("some-input", "bar").header("some-other-input", "bar2")
+                .get("/custom/req")
+                .then().statusCode(200).body(Matchers.containsString("/custom/req-bar-bar2-null")).extract().headers();
         assertThat(responseHeaders.getValues("java-method")).containsOnly("filters");
         Assertions.assertEquals(3, AssertContainerFilter.COUNT.get());
         assertThat(responseHeaders.getValues("very")).isEmpty();
 
-        responseHeaders = RestAssured.given().header("some-input", "bar").get("/custom/metal")
-                .then().statusCode(200).body(Matchers.containsString("/custom/metal-bar-metal")).extract().headers();
+        responseHeaders = RestAssured.given().header("some-input", "bar").header("some-other-input", "bar2")
+                .get("/custom/metal")
+                .then().statusCode(200).body(Matchers.containsString("/custom/metal-bar-bar2-metal")).extract().headers();
         assertThat(responseHeaders.getValues("very")).containsOnly("heavy");
     }
 }

--- a/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/ResteasyReactiveDotNames.java
+++ b/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/ResteasyReactiveDotNames.java
@@ -273,6 +273,9 @@ public final class ResteasyReactiveDotNames {
     public static final DotName RESTEASY_REACTIVE_CONTAINER_REQUEST_CONTEXT = DotName
             .createSimple("org.jboss.resteasy.reactive.server.spi.ResteasyReactiveContainerRequestContext");
 
+    public static final DotName RESTEASY_REACTIVE_SERVER_REQUEST_CONTEXT = DotName
+            .createSimple("org.jboss.resteasy.reactive.server.spi.ServerRequestContext");
+
     public static final DotName WITH_FORM_READ = DotName
             .createSimple("org.jboss.resteasy.reactive.server.WithFormRead");
 

--- a/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/exceptionmappers/ServerExceptionMapperGenerator.java
+++ b/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/exceptionmappers/ServerExceptionMapperGenerator.java
@@ -9,6 +9,7 @@ import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNa
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.RESPONSE;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.REST_RESPONSE;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.SERVER_EXCEPTION_MAPPER;
+import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.SERVER_REQUEST_CONTEXT;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.UNI;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.URI_INFO;
 import static org.jboss.resteasy.reactive.server.processor.generation.multipart.GeneratorUtils.paramHandleFromReqContextMethod;
@@ -569,6 +570,8 @@ public final class ServerExceptionMapperGenerator {
                         ofMethod(ResteasyReactiveRequestContext.class.getName(), "getContainerRequestContext",
                                 ContainerRequestContextImpl.class),
                         contextHandle);
+            } else if (SERVER_REQUEST_CONTEXT.equals(paramDotName)) {
+                targetMethodParamHandles[i] = contextHandle;
             } else if (URI_INFO.equals(paramDotName)) {
                 paramHandleFromReqContextMethod(mc, contextHandle, targetMethodParamHandles, i,
                         "getUriInfo",

--- a/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java
+++ b/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java
@@ -8,6 +8,7 @@ import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNa
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.RESOURCE_INFO;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.RESPONSE;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.REST_RESPONSE;
+import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.SERVER_REQUEST_CONTEXT;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.THROWABLE;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.UNI;
 import static org.jboss.resteasy.reactive.common.processor.ResteasyReactiveDotNames.URI_INFO;
@@ -330,6 +331,8 @@ final class CustomFilterGenerator {
             } else if (ResteasyReactiveServerDotNames.QUARKUS_REST_CONTAINER_REQUEST_CONTEXT.equals(paramDotName)) {
                 targetMethodParamHandles[i] = filterMethod.checkCast(filterMethod.getMethodParam(0),
                         ResteasyReactiveContainerRequestContext.class);
+            } else if (SERVER_REQUEST_CONTEXT.equals(paramDotName)) {
+                targetMethodParamHandles[i] = rrReqCtxHandle;
             } else if (URI_INFO.equals(paramDotName)) {
                 GeneratorUtils.paramHandleFromReqContextMethod(filterMethod, rrReqCtxHandle, targetMethodParamHandles,
                         i,

